examples/module1/lesson1/solver/utils/functions.ts

```diff
@@ -0,0 +1,12 @@
+export const add = (firstVal: number, secondVal: number) => firstVal + secondVal;
+
+export const subtract = (firstVal: number, secondVal: number) => firstVal - secondVal;
+
+export const multiply = (firstVal: number, secondVal: number) => firstVal * secondVal;
+
+export const divide = (firstVal: number, secondVal: number) => {
+  if (secondVal === 0) {
+    return 'Cannot divide by zero';
```

Comment:
to prosi się o kłopoty. funkcja divide nie powinna zwracać błędu walidacji jako rezultat działania funkcji tylko jak już to rzucać wyjątkiem. Łamiesz tutaj single responsibility oraz dodatkowo generujesz sobie nadmiarowe problemy w dalszej implementacji kodu. Powiedzmy że np chcesz chainować operacje, każda operacja aby mogła się dalej odbyć musi zwracać number. Jak nagle jedna z operacji zwraca number | string to musisz tworzyć nadmierną logikę aby upewnić się że wynikiem operacji jest number. To jeden z przykładów gdzie błąd powstanie ale może być ich dużo więcej.